samples/gitchurn/src/gitChurnMain/kotlin/GitChurn.kt

```diff
@@ -35,33 +35,44 @@ fun main(args: Array<String>) {
 private fun calculateChurn(workDir: String, limit: Int) {
     println("Opening…")
     val repository = git.repository(workDir)
-    val map = mutableMapOf<String, Int>()
+    data class WhoAndWhere(val authorEmail: String = "", val filePath: String = "")
+    val modificationsByAuthor = mutableMapOf<WhoAndWhere, Int>()
     var count = 0
-    val commits = repository.commits()
-    val limited = commits.take(limit)
+    val commits = repository.myCommits()
+    val commitsLimit = commits.take(limit)
     println("Calculating…")
-    limited.forEach { commit ->
+    commitsLimit.forEach { commit ->
         if (count % 100 == 0)
-            println("Commit #$count [${commit.time.format()}]: ${commit.summary}")
+            println("Commit #$count [${commit.time.format()}] by ${commit.author.name!!.toKString()}: ${commit.summary}")
 
         commit.parents.forEach { parent ->
             val diff = commit.tree.diff(parent.tree)
             diff.deltas().forEach { delta ->
-                val path = delta.newPath
-                val n = map[path] ?: 0
-                map.put(path, n + 1)
+                val path = if (delta.newPath == delta.newPath.substringBefore('/'))
+                    delta.newPath
+                else
+                    delta.newPath.substringBefore('/') + "/"
+                val n = modificationsByAuthor[WhoAndWhere(parent.author.email!!.toKString().toLowerCase(), path)] ?: 0
+                modificationsByAuthor[WhoAndWhere(parent.author.email!!.toKString().toLowerCase(), path)] = n + 1
             }
             diff.close()
             parent.close()
         }
         commit.close()
         count++
     }
-    println("Report:")
-    map.toList().sortedByDescending { it.second }.take(10).forEach {
-        println("File: ${it.first}")
-        println("      ${it.second}")
-        println()
+    println("Named Report:")
+    modificationsByAuthor.toList().groupBy { it.first.authorEmail }.values.forEach {
+        println("Author: ${it[0].first.authorEmail}")
+        var changedFilesSum = 0
+        it.sortedByDescending { it.second }.forEach(fun(it: Pair<WhoAndWhere, Int>) {
```

Comment:
What's the motivation behind using `fun(...) {}` syntax instead of plain lambda?

Reply:
IDEA told me, that there was two different `it`, one from the outer cycle, and one from the inner one. To make it more straightforward, it recommended to declare my `it` this way. Now I see, that I need just to rewrite this part with more named variables.

Reply:
Btw, you can specify lambda parameter to have name other than `it`, e.g.
```Kotlin
{ parameter ->
}
```
You can also achieve that by using Alt+Enter on `it`.

On the other hand, using `fun(it: Pair<WhoAndWhere, Int>) {` doesn't seem to avoid shadowing one `it` by another.